Assets/2_MainScene/S2_Scripts/S2_BossScripts/S2_BossManager.cs

```diff
@@ -72,6 +139,38 @@ public void EndBoss()
         shieldTest.text = shield.ToString();
     }
 
+    public void DisplayWarning()
+    {
+        bossCanvas.SetActive(true);
+        healthBar.gameObject.SetActive(false);
+        shieldBar.gameObject.SetActive(false);
+        warningObject.SetActive(true);
+        warningText.SetActive(true);
+        StartCoroutine(WarningFlash());
+    }
+
+    WaitForSeconds waitTime = new WaitForSeconds(0.5f);
+    IEnumerator WarningFlash()
```

Comment:
Legitimately not sure but would would it be less expensive to use an InvokeRepeating with a counter?

> public void DisplayWarning() { InvokeRepeating(nameof(WarningFlash), 0.5f, 0.5f); }

```
int counter = 0;
void WarningFlash() {
     if(counter >= 5) { 
          CancelInvoke(nameof(WarningFlash));
          Return;
      }
     
     wanringObject.SetActive(!warningObject.activeSelf);
     warningText.SetActive(!warningText.activeSelf);
     
     counter++;
}
```

Also not sure if a canceled Invoke finishes it's iteration so the Return may not be needed

Also Also, is it intended to have the final iteration turn the warnings on or are they removed another way?

Reply:
I've looked some more into this and the best sources i've found seem to say that invokerepeating is generally slightly better performance wise than coroutines. However there are exceptions to this and if we run into specific areas of poor performance, we can look into coroutines vs invokes at a case by case basis.

The return is needed cause the cancel invoke will only prevent it from being called further and would still finish the iteration. 

In this case, there is something else turning everything off.